pkg/util/file_indexer.go

```diff
@@ -83,6 +86,57 @@ func resolveFilePath(directory string) (string, error) {
 	return directory, nil
 }
 
+// gitignoreFilePath gives the filepath of the .gitignore file in the context
+func gitIgnoreFilePath(directory string) (string, error) {
+	_, err := os.Stat(filepath.Join(directory))
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(directory, ".gitignore"), nil
+}
+
+// addOdoFileIndex adds odo-file-index.json to .gitignore
+func addOdoFileIndex(gitIgnoreFile string) error {
+	var data []byte
+	file, err := os.OpenFile(gitIgnoreFile, os.O_APPEND|os.O_RDWR, 0600)
+	if err != nil {
+		return errors.Wrap(err, "failed to open .gitignore file")
+	}
+	defer file.Close()
+
+	if data, err = ioutil.ReadFile(gitIgnoreFile); err != nil {
+		return errors.Wrap(err, "failed reading data from .gitignore file")
+	}
+	// check whether .odo/odo-file-index.json is already in the .gitignore file
+	if !strings.Contains(string(data), filepath.Join(fileIndexDirectory, fileIndexName)) {
+		if _, err := file.WriteString("\n" + filepath.Join(fileIndexDirectory, fileIndexName)); err != nil {
+			return errors.Wrapf(err, "failed to Add %v to .gitignore file", fileIndexName)
+		}
+	}
+	return nil
+}
+
+// Check .gitignore file exists or not, if not then create it
+func checkGitIgnoreFile(directory string) (string, error) {
+
+	gitIgnoreFile, err := gitIgnoreFilePath(directory)
+	if err != nil {
+		return "", err
+	}
+	// err checks the existence of .gitignore and then creates if does not exists
+	_, err = os.Stat(gitIgnoreFile)
+
+	if err != nil {
```

Comment:
The err can be other type too and thus may not indicate that the file doesn't exist. Please check the err to know the type like below

```
if _, err := os.Stat(filePath); os.IsNotExist(err) {
           // logic here
	}
```